syne_tune/optimizer/schedulers/multiobjective/moasha.py

```diff
@@ -54,220 +55,122 @@ class MOASHA(LegacyTrialScheduler):
         ]
-        self._num_stopped = 0
-        self._metrics = metrics
-        self._mode = mode
-        if isinstance(self._mode, List):
-            self._metric_op = {
-                metric: 1 if mode == "min" else -1
-                for metric, mode in zip(metrics, self._mode)
-            }
-        else:
-            if self._mode == "min":
-                self._metric_op = dict(zip(self._metrics, [1.0] * len(self._metrics)))
-            elif self._mode == "max":
-                self._metric_op = dict(zip(self._metrics, [-1.0] * len(self._metrics)))
-        self._time_attr = time_attr
+        self.num_stopped = 0
+        self.metrics = metrics
+        self.metric_multiplier = 1 if self.do_minimize else -1
+        self.time_attr = time_attr
 
     def metric_names(self) -> List[str]:
-        return self._metrics
+        return self.metrics
 
     def metric_mode(self) -> str:
-        return self._mode
+        return "min" if self.do_minimize else "max"
 
-    def _suggest(self, trial_id: int) -> Optional[TrialSuggestion]:
+    def suggest(self) -> Optional[TrialSuggestion]:
         config = {
             k: v.sample() if hasattr(v, "sample") else v
             for k, v in self.config_space.items()
         }
         return TrialSuggestion.start_suggestion(config)
 
     def on_trial_add(self, trial: Trial):
-        sizes = np.array([len(b._rungs) for b in self._brackets])
+        sizes = np.array([len(b.rungs) for b in self.brackets])
         probs = np.e ** (sizes - sizes.max())
         normalized = probs / probs.sum()
-        idx = np.random.choice(len(self._brackets), p=normalized)
-        print(f"adding trial {trial.trial_id}")
-        self._trial_info[trial.trial_id] = self._brackets[idx]
+        idx = np.random.choice(len(self.brackets), p=normalized)
+        self.trial_info[trial.trial_id] = self.brackets[idx]
 
     def on_trial_result(self, trial: Trial, result: Dict[str, Any]) -> str:
-        self._check_metrics_are_present(result)
-        if result[self._time_attr] >= self._max_t:
+        self.check_metrics_are_present(result)
+        if result[self.time_attr] >= self.max_t:
             action = SchedulerDecision.STOP
         else:
-            bracket = self._trial_info[trial.trial_id]
-            metrics = self._metric_dict(result)
+            bracket = self.trial_info[trial.trial_id]
+            metrics = self.metric_dict(result)
             action = bracket.on_result(
                 trial_id=trial.trial_id,
-                cur_iter=result[self._time_attr],
+                cur_iter=result[self.time_attr],
                 metrics=metrics,
             )
         if action == SchedulerDecision.STOP:
-            self._num_stopped += 1
+            self.num_stopped += 1
         return action
 
-    def _metric_dict(self, reported_results: Dict[str, Any]) -> Dict[str, Any]:
+    def metric_dict(self, reported_results: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Fine with me but is there a particular reason to remove the underscore in private methods?